### DISCUSSION TITLE
Is there any variable environment to get the server url?

### DISCUSSION THREAD

**Discussion by mateusfg7** [vercel/next.js#18857] [Help] [answered]
I'm using Next's own servless API, but to be able to fetch this API I need to use the absolute URL, if there is an environment variable with this information, how can I make this URL be relative to the environment that the server is running, between dev server (https://localhost:3000) and production (DNS).

![image](https://user-images.githubusercontent.com/40613276/98293682-7fce2c00-1f8d-11eb-827d-75f4d16b8b39.png)

_fetch_
```js
const currencyInfoApiResponse = await fetch(
  'http://localhost:3000/api/crypto'
 )
const currencyInfoApiJson: CurrencyInfoResponse = await currencyInfoApiResponse.json()
```

i want to do something like this
```js
const currencyInfoApiResponse = await fetch(
  `${process.env.HOST}/api/crypto`
 )
const currencyInfoApiJson: CurrencyInfoResponse = await currencyInfoApiResponse.json()
```

**Comment by lfades** [comment, marked as answer]
Next.js doesn't have a way of knowing that. Try the following instead:

- Use relative paths if you're making a request from the browser
- If you're using API Routes, or one of the Next.js data fetching methods like `getStaticProps` and `getServerSideProps`, import the code directly instead of making an API call, Next.js will make sure to remove any Node.js from the browser build.

If getting access to the server URL is a must, then you need to set the environment variable yourself before starting the Next.js app. If you're using Vercel you can enable [system environment variables](https://vercel.com/docs/build-step?query=git%2520environment%2520variables#system-environment-variables) to get access to the URL
